dip-0013.md

```diff
@@ -0,0 +1,163 @@
+
+Identity Funding keys can either be for a registration or for a topup. We will create for each
+wallet two sets of secp256k1 keys. Funding transactions using these keys will be perceived as
+identical on the network. However by separating these keys based on two derivation paths, wallets
+can keep track of the motivation behind each credit funding transaction.
+
+Identity Funding keys used for registration are the second sub-feature for identities. They are
+always derived as ECDSA keys.
+
+We define the following levels for them:
+
+`m / purpose' / coin_type' / feature' / sub feature' / identity index`
+
+Sub feature is set to `1'`.
+
+As there can only be one registration funding per identity, there is no point to derive a level
+deeper.
+
+The index of the identity for funding need not be the same as the index used in authentication keys.
+The relationship between the two can be retrieved in information from Dash Platform upon successful
+identity registration.
+
+If a wallet recovers a registration funding transaction without an associated identity using this
+funding, the wallet can safely assume that the identity was either in the process of being
+registered when an error occurred or the process was terminated by the user. The wallet should offer
+the user the ability to resume the process.
+
+It is recommended that five address hashes of these keys be added to the bloom filter sent to peers.
+If four identities have been found, the filter should be regenerated with five more. Address hashes
+must use HASH160 of the public key which is one round of SHA256 followed by one round of RIPEMD160.
+
+# Identity Top Up Funding keys
+
+Identity Funding keys used for topup are the third sub-feature for identities. They are always
+derived as ECDSA keys.
+
+We define the following levels for them:
+
+`m / purpose' / coin_type' / feature' / sub feature' / funding index /`
+
+Sub feature is set to `2'`.
+
+We should note that unlike Identity Registration keys, Top Up Funding keys do not reference an
+identity and can be used to top up any identity. It is useful to point out that a registration
+funding key is a source of entropy in the unique identifier of the identity it will fund, and this
+is obviously not the case for top up funding keys.
+
+If a wallet recovers an unused top up funding transaction, the wallet should attempt to reuse it.
+This is done in a few steps. First, the wallet must recognise the key used in a returned identity
+funding transaction as a top up funding key. Then the wallet must query Dash Platform to check if
+the top up transaction has already been used and the balance transferred into credits. If this is
+not the case, the wallet can then perform an identity top up transition as defined in DIP11 using
+the top up funding key.
+
+It is recommended that 30 address hashes of these keys be added to the bloom filter sent to peers.
+If 25 identities have been found, the filter should be regenerated with 30 more.
```

Comment:
This has been a while since I've worked on this part so I hope my memory is good enough. Since the bloom filter is made for the peer in question and since clients switch peers when changing bloom filters (to maintain privacy) it is essentially regenerated.

Reply:
And you found an error @UdjinM6 ! it's not 25 identities, it's 25 addresses. I'll fix this.